src/main/compiler/CompilerScope.ts

```diff
@@ -274,35 +276,44 @@ export class CompilerScope {
     }
 
     private emitEvalSync(node: NodeView, resSym: string) {
-        const code = node.getProp('code')?.value ?? '';
-        this.code.block(`const $p = {`, `}`, () => {
-            const prop = node.getProp('args');
-            if (prop) {
-                this.emitProp(prop);
-            }
-        });
-        const args = node.getProp('args')?.getEntries() ?? [];
-        const argList = args.map(_ => _.key).join(',');
-        const argVals = args.map(_ => `$p.args[${JSON.stringify(_.key)}]`).join(',');
-        this.code.block(`${resSym} = ((${argList}) => {`, `})(${argVals})`, () => {
-            this.code.line(code);
-        });
+        this.emitEvalLike(node, resSym, false, 'code', 'args');
     }
 
     private emitEvalAsync(node: NodeView, resSym: string) {
-        const code = node.getProp('code')?.value ?? '';
+        this.emitEvalLike(node, resSym, true, 'code', 'args');
+    }
+
+    private emitAI(node: NodeView, resSym: string) {
+        const async = node.isAsync();
+        this.emitEvalLike(node, resSym, async, 'code', 'inputs');
+    }
+
+    private emitEvalLike(
+        node: NodeView,
+        resSym: string,
+        async: boolean,
```

Comment:
could these be `isAsync` or something other than just `async`?

Reply:
Ah I see what you mean; the param name can be changed, but not the field name of the `metadata` and `moduleSpec` because those are already in use (and part of public interface)

Reply:
ah ok, it's no biggy anyway, was just a little confusing at first glance 👍